blackjack/console.py

```diff
@@ -0,0 +1,105 @@
+from core import *
+
+
+def show_cards(state: GameState):
+    print("")
+    print(f'Player 1: {render_hand(state.player_hand)}')
+    print(f'The House: {render_dealer(state.dealer_hand)}\n')
+
+
+def display_winner(winner: Win, state: GameState):
+    player_total = hand_total(state.player_hand)
+    house_total = hand_total(state.dealer_hand)
+    if winner is Win.Push:
+        print("")
+        print(f'The House: {render_hand(state.dealer_hand)}= {house_total}')
+        print(f'Player 1: {render_hand(state.player_hand)}= {player_total} \n PUSH!')
+        print("")
+    if winner is Win.Player_bj:
+        print("")
+        print(f'The House: {render_hand(state.dealer_hand)}= {house_total}')
+        print(f'BLACKJACK! {render_hand(state.player_hand)}= {player_total} \nYOU WIN!')
+        print("")
+    if winner is Win.House_bj:
+        print("")
+        print(f'Player 1: {render_hand(state.player_hand)}= {player_total}')
+        print(f'BLACKJACK! {render_hand(state.dealer_hand)}= {house_total} \nHouse wins!')
+        print("")
+    if winner is Win.Player_bust:
+        print("")
+        print(f'The House: {render_hand(state.dealer_hand)}= {house_total}')
+        print(f'You Bust! {render_hand(state.player_hand)}= {player_total} \nHouse Wins!')
+        print("")
+    if winner is Win.House_bust:
+        print("")
+        print(f'Player 1: {render_hand(state.player_hand)}= {player_total}')
+        print(f'House Busts: {render_hand(state.dealer_hand)}= {house_total} \nYou Win!')
+        print("")
+    if winner is Win.House_win:
+        print("")
+        print(f'The House: {render_hand(state.dealer_hand)}= {house_total}')
+        print(f'Player 1: {render_hand(state.player_hand)}= {player_total} \n House Wins!')
+        print("")
+    if winner is Win.Player_win:
+        print("")
+        print(f'The House: {render_hand(state.dealer_hand)}= {house_total}')
+        print(f'Player 1: {render_hand(state.player_hand)}= {player_total} \n YOU Win!')
+        print("")
+
+
+def get_hit_or_stay(state: GameState) -> Play:
+    total = hand_total(state.player_hand)
+    s = input(f'Your Total is {total}, would you like to Hit? ("y" or "n"): ')
+    play = parse_play(s)
+    if play:
+        return play
+    print("Please choose 'y' or 'n'.")
+    return get_hit_or_stay(state)
+
+
+def play_again():
+    leaving = True
+    while leaving:
+        a = input("Would you like to play again? ('y' or 'n'): ")
+        print("")
+        if a == 'y':
+            leaving = False
+            main()
```

Comment:
It would be better to not call `main()` again; ideally, `main()` is only called in the `if __name__` block. So you likely need to refactor things into another function.

Reply:
Also, note that by calling `main()` you are resetting the state of the deck...this is not right. You should continue to re-use the deck until you need to reshuffle, then you should create a new deck and shuffle it.

Reply:
Resolved. I believe this is where the mystery bug was like you said. it seems to be running a bit smoother.